packages/core/lib/get-permissions.ts

```diff
@@ -0,0 +1,48 @@
+import { AppState, ComponentData, Config } from "../types/Config";
+
+export const getPermissions = ({
+  permissions,
+  selectedItem,
+  state,
+  config,
+}: {
+  permissions: string[];
+  selectedItem: ComponentData | undefined;
+  state: AppState;
+  config: Config;
+}) => {
+  let componentPermissions =
+    selectedItem && initialPermissions({ selectedItem, config, state });
+
+  const computedPermissions = componentPermissions
+    ? Object.keys(componentPermissions)
+        .filter((key) => permissions.includes(key))
+        .reduce((obj, key) => {
+          obj[key] = componentPermissions[key] ?? null;
+          return obj;
+        }, {} as { [key: string]: boolean | null })
+    : null;
+
+  return computedPermissions;
+};
+
+export const initialPermissions = ({
```

Comment:
Prefer to keep functions as verbs -

```suggestion
export const getInitialPermissions = ({